Assets/_BForBoss/_Utility/Editor/PropertyDrawers/ResolverPropertyDrawer.cs

```diff
@@ -0,0 +1,369 @@
+                    {
+                        Transform parent = mb.transform.parent;
+                        
+                        while (parent != null)
+                        {
+                            gameObjects.Add(parent.gameObject);
+                            parent = parent.parent;
+                        }
+                    }
+                    else
+                    {
+                        components = mb.GetComponentsInParent(fieldType, _includeInactiveGameObjects)
+                            .Where(component => component.gameObject != mb.gameObject).ToArray();
+                    }
+                    
+                    break;
+                }
+                case ResolveType.FromSelf:
+                {
+                    if (isGameObjectType)
+                    {
+                        gameObjects = new List<GameObject>{mb.gameObject};
+                    }
+                    else
+                    {
+                        components = mb.GetComponents(fieldType);
+                    }
+                    
+                    break;
+                }
+            }
+            
+            void OnItemSelected(int componentID)
+            {
+                property.serializedObject.Update();
+                property.objectReferenceInstanceIDValue = componentID;
+                property.serializedObject.ApplyModifiedProperties();
+            }
+
+            if (isGameObjectType)
+            {
+                if (gameObjects.IsNullOrEmpty())
+                {
+                    return;
+                }
+
+                if (gameObjects.Count > 1)
+                {
+                    var dropdown = new GameObjectResolverDropdown(gameObjects, OnItemSelected, new AdvancedDropdownState());
+                    dropdown.Show(_childResolveRect);
+                }
+                else
+                {
+                    property.serializedObject.Update();
+                    property.objectReferenceValue = gameObjects[0];
+                    property.serializedObject.ApplyModifiedProperties();
+                }
+            }
+            else
+            {
+                if (components.IsNullOrEmpty())
+                {
+                    return;
+                }
+
+                if (components.Length > 1)
+                {
+                    var dropdown = new ComponentResolverDropdown(components, OnItemSelected , new AdvancedDropdownState());
+                    dropdown.Show(_childResolveRect);
+                }
+                else
+                {
+                    property.serializedObject.Update();
+                    property.objectReferenceValue = components[0];
+                    property.serializedObject.ApplyModifiedProperties();
+                }
+            }
+        }
+
+        private void InitializeContent()
+        {
+            _childResolveContent = new GUIContent("C", "Get Component from Children");
```

Comment:
"Get Component(s) from Children"

Reply:
Well you are only resolving to one component, so shouldn't we use the singular in this case?

Reply:
oh in my headd, you are showing all the components in the children. but what you wrote is fine too.